kafka-plugins-common/src/main/java/co/cask/hydrator/plugin/batch/source/KafkaBatchConfig.java

```diff
@@ -0,0 +1,340 @@
+package co.cask.hydrator.plugin.batch.source;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.data.format.FormatSpecification;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.format.RecordFormats;
+import co.cask.hydrator.common.KeyValueListParser;
+import co.cask.hydrator.common.ReferencePluginConfig;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Config properties for the plugin.
+ */
+public class KafkaBatchConfig extends ReferencePluginConfig {
+
+  @Description("Kafka topic to read from.")
+  @Macro
+  private String topic;
+
+  @Description("List of Kafka brokers specified in host1:port1,host2:port2 form. For example, " +
+    "host1.example.com:9092,host2.example.com:9092.")
+  @Macro
+  private String kafkaBrokers;
+
+  @Description("A directory path to store the latest Kafka offsets. " +
+    "A file named with the pipeline name will be created under the given directory.")
+  @Macro
+  @Nullable
+  private String offsetDir;
```

Comment:
widget files and reference docs need to be updated to include this new property.